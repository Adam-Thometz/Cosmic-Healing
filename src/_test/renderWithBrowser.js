import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

function withProvider({ children, initialRoutes }) {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
)
};

function renderWithBrowser(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithBrowser;