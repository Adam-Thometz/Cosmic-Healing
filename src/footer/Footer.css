.Footer {
  background-image: 
    url(../_media/backgrounds/black-lines.png),
    url(../_media/backgrounds/black.png);
  background-size: 100% 100%, 100% 100%;
  background-repeat: no-repeat;
  height: 330px;
  display: flex;
}

.Footer-menu {
  display: flex;
  flex-direction: column;
  text-align: left;
  height: 50%;
  margin: 3%;
  justify-content: top;
}