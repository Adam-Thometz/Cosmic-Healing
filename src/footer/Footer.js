import React from 'react';
import SmallCaption from '../_components/small-caption/SmallCaption';
import Link from '../_components/link/Link';

import './Footer.css';

const Footer = () => {
  return (
    <footer className='Footer'>
      <section className='Footer-menu'>
        <SmallCaption color='#FBAD7D' text="VESTA HEALING" />
        <Link link='/'>Home</Link>
        <Link link='/about-me'>About Me</Link>
        <Link link='/services'>Services</Link>
        <Link link='/services#faq'>FAQ</Link>
      </section>
      <section className='Footer-menu'>
        <SmallCaption color='#FBAD7D' text="COMPANY" />
        <Link link='/'>Contact Us</Link>
      </section>
    </footer>
  );
};

export default Footer;