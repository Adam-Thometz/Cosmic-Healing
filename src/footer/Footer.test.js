import React from "react";
import renderWithBrowser from '../_test/renderWithBrowser';
import Footer from './Footer';

describe('Footer component', () => {
  it('renders without crashing', () => {
    renderWithBrowser(<Footer />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithBrowser(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});