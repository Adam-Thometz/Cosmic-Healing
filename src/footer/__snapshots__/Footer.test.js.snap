// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="Footer"
  >
    <section
      class="Footer-menu"
    >
      <small
        class="SmallCaption"
        style="color: rgb(251, 173, 125);"
      >
        VESTA HEALING
      </small>
      <a
        class="Link tertiary"
        href="/"
      >
        Home
      </a>
      <a
        class="Link tertiary"
        href="/about-me"
      >
        About Me
      </a>
      <a
        class="Link tertiary"
        href="/services"
      >
        Services
      </a>
      <a
        class="Link tertiary"
        href="/services#faq"
      >
        FAQ
      </a>
    </section>
    <section
      class="Footer-menu"
    >
      <small
        class="SmallCaption"
        style="color: rgb(251, 173, 125);"
      >
        COMPANY
      </small>
      <a
        class="Link tertiary"
        href="/"
      >
        Contact Us
      </a>
    </section>
  </footer>
</DocumentFragment>
`;
